admin/src/pages/Reports/components/ReportsTableRow/index.tsx

```diff
@@ -0,0 +1,351 @@
+  updateReports,
+  selectedReports,
+  onSelectionChange,
+  isChecked,
+  handleItemSelectionChange,
+}) => {
+  const {
+    id,
+    reports: reportedReports,
+    related: {id: commentID},
+  } = item;
+
+  const [blockButtonsDisabled, setBlockButtonsDisabled] = useState(false);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [blockButtonsVisible, setBlockButtonsVisible] = useState(false);
+
+  const {formatDate} = useIntl();
+  const toggleNotification = useNotification();
+  const queryClient = useQueryClient();
+  const {lockApp, unlockApp} = useOverlayBlocker();
+
+  const onSuccess =
+    (stateAction = () => {}, indalidate = true) =>
+    async () => {
+      if (indalidate) {
+        await queryClient.invalidateQueries('get-data');
+      }
+      stateAction(false);
+      unlockApp();
+    };
+
+  const onError = (err) => {
+    handleAPIError(err, toggleNotification);
+  };
+
+  const blockItemMutation = useMutation(blockItem, {
+    onSuccess: onSuccess(),
+    onError,
+    refetchActive: false,
+  });
+
+  const unblockItemMutation = useMutation(unblockItem, {
+    onSuccess: onSuccess(),
+    onError,
+    refetchActive: false,
+  });
+
+  const blockItemThreadMutation = useMutation(blockItemThread, {
+    onSuccess: onSuccess(),
+    onError,
+    refetchActive: false,
+  });
+  const unblockItemThreadMutation = useMutation(unblockItemThread, {
+    onSuccess: onSuccess(),
+    onError,
+    refetchActive: false,
+  });
+
+  const isNotResolved = (entry) => !entry.resolved;
+
+  const handleClickResolve = async (reportId) => {
+    if (canReviewReports) {
+      lockApp();
+      const item = await mutation.mutateAsync({
+        id: commentID,
+        reportId,
+      });
+      if (item) {
+        const updatedReports = reports.map((_) => ({
+          ..._,
+          resolved: reportId === _.id ? true : _.resolved,
+        }));
+        updateReports(updatedReports);
+        onSelectionChange(selectedReports.filter((_) => _ !== reportId));
+      }
+    }
+  };
+
+  const handleBlockItemClick = async (id) => {
+    if (canModerate) {
+      lockApp();
+      blockItemMutation.mutate(id);
+    }
+  };
+
+  const handleUnblockClick = async (id) => {
+    if (canModerate) {
+      lockApp();
+      unblockItemMutation.mutate(id);
+    }
+  };
+
+  const handleBlockItemThreadClick = async (id) => {
+    if (canModerate) {
+      lockApp();
+      blockItemThreadMutation.mutate(id);
+    }
+  };
+
+  const handleUnblockThreadClick = async (id) => {
+    if (canModerate) {
+      lockApp();
+      unblockItemThreadMutation.mutate(id);
+    }
+  };
+
+  const isLoading =
+    blockItemMutation.isLoading || blockItemThreadMutation.isLoading;
+  const openReports = reportedReports?.filter((_) => !_.resolved);
+  const hasReports = !isEmpty(openReports);
+  const reviewFlowEnabled =
+    canAccessReports && hasReports && !(item.blocked || item.blockedThread);
+
+  const handleClick = () => onClick();
```

Comment:
Could we do it like this?
```suggestion
  const handleClick = onClick;
```